evolver/base.py

```diff
@@ -77,17 +206,59 @@ def _setup_logger(self):
         ch.setFormatter(formatter)
         self.logger.addHandler(ch)
 
+    @classmethod
+    def __get_pydantic_core_schema__(cls, *args, **kwargs):
+        return cls.Config.__get_pydantic_core_schema__(*args, **kwargs)
 
-class ConfigDescriptor(pydantic.BaseModel):
-    classinfo: pydantic.ImportString
-    config: dict = {}
+    @classmethod
+    def __get_pydantic_json_schema__(cls, *args, **kwargs):
+        return cls.Config.__get_pydantic_json_schema__(*args, **kwargs)
 
-    def create(self, **kwargs):
-        """ Create an instance of classinfo from a config.
+    @classmethod
+    def fqn(cls):
+        return f"{cls.__module__}.{cls.__qualname__}"
 
-            Args:
-                kwargs (:obj:`dict`): Key-value pairs used to override contents of ``self.config``.
+    @property
+    def config(self) -> dict:
```

Comment:
A significant change from #37 - this no longer returns a pedantic model but instead a ``dict``. Doing so makes this easier to automatically convert back and forth from a ``ConfigDescriptor`` as pydantic will use the attributes from the object being passed in and therefore the names must match.